cmake/HPX_SetupTarget.cmake

```diff
@@ -188,7 +188,11 @@ function(hpx_setup_target target)
   endif()
 
   if(NOT target_NOLIBS)
-    set(hpx_libs hpx)
+    if(HPX_WITH_DYNAMIC_HPX_MAIN AND ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux"))
```

Comment:
Shouldn't this happen for executables only?

Reply:
@hkaiser if we limit it to Executables only, then some shared objects that link to it in phylanx will throw a linker error for not defined `__real___libc_start_main`. Providing this flag to non-executables will not produce unexpected results.